### ISSUE TITLE
FMI export issue for Modelica.Fluid.Examples.PumpingSystem

### ISSUE THREAD

**Issue by lochel** [OpenModelica/OpenModelica#13272]
### Description

When running an FMU (created using Modelica.Fluid.Examples.PumpingSystem) in Python with OMSimulator, the following errors occur:

```
status:  [correct] ok
status:  [correct] ok
error:   [fmi2logger] D:/TestFMU/Sample/test_tmp/test-hrobx1ba/temp/0001_A (logStatusError): C:/Users/DEMO/AppData/Roaming/.openmodelica/libraries/Modelica 4.0.0+maint.om/Media/Water/IF97_Utilities.mo:2900: IF97 medium function tsat called with too low pressure
         p = -1.70619e+08 Pa <= 611.657 Pa (triple point pressure)
error:   [fmi2logger] D:/TestFMU/Sample/test_tmp/test-hrobx1ba/temp/0001_A (logFmi2Call): fmi2CompletedIntegratorStep: terminated by an assertion.
error:   [fmi2logger] D:/TestFMU/Sample/test_tmp/test-hrobx1ba/temp/0001_A (logStatusError): C:/Users/DEMO/AppData/Roaming/.openmodelica/libraries/Modelica 4.0.0+maint.om/Media/Water/IF97_Utilities.mo:2900: IF97 medium function tsat called with too low pressure
         p = -1.70619e+08 Pa <= 611.657 Pa (triple point pressure)
status:  [correct] ok
status:  [correct] ok
status:  [correct] ok
LOG_ASSERT        | warning | [C:/Users/DEMO/AppData/Roaming/.openmodelica/libraries/Modelica 4.0.0+maint.om/Fluid/Interfaces.mo:1027:9-1032:39:writable]
|                 | |       | The following assertion has been violated at time 0.000000
|                 | |       | ((pipe.flowModel.m_flows[1] >= 0.0 and pipe.flowModel.m_flows[1] <= 1e5)) --> "Variable violating min/max constraint: 0.0 <= pipe.flowModel.m_flows[1] <= 1e5, has value: -141.424"
info:    No result file will be created
status:  [correct] ok
Minutes: 0; Seconds: 1
Traceback (most recent call last):
OSError: exception: access violation reading 0x0000000000000000
warning: temp directory "D:\TestFMU\Sample\test_tmp\test-hrobx1ba" couldn't be removed
         remove_all: Access is denied.: "D:\TestFMU\Sample\test_tmp\test-hrobx1ba"
info:    1 warnings
info:    3 errors
```

Software Versions
* OpenModelica: v1.24.0-dev-20-gde46d5b2f7 (64-bit)
* OMSimulator-V2.1.1Post257
* Operating System: Windows 11 (64-bit)

---

Originally reported by @Primkins in OpenModelica/OMSimulator#1294.

**Comment by casella** [comment]
@Primkins have you tried with the latest version 1.24.2?

**Comment by Primkins** [comment]
@casella we have tested the issue in the latest version 1.24.2. issue status remains same.

**Comment by adrpo** [comment]
We need more information on how you tested this. How do you call OMSimulator?
I get this with OM v1.25.0-dev-115-g6254a2ee46 for export and OMSimulator v2.1.2.post8-gf48ed62-mingw-notlm-debug.
```
c:\temp>"%OPENMODELICAHOME%"\bin\OMSimulator PumpingSystem.fmu
LOG_SOLVER        | info    | CVODE linear multistep method CV_BDF
LOG_SOLVER        | info    | CVODE maximum integration order CV_ITER_NEWTON
LOG_SOLVER        | info    | CVODE use equidistant time grid YES
LOG_SOLVER        | info    | CVODE Using relative error tolerance 1.000000e-06
LOG_SOLVER        | info    | CVODE Using dense internal linear solver SUNLinSol_Dense.
LOG_SOLVER        | info    | CVODE Use internal dense numeric jacobian method.
LOG_SOLVER        | info    | CVODE uses internal root finding method NO
LOG_SOLVER        | info    | CVODE maximum absolut step size 0
LOG_SOLVER        | info    | CVODE initial step size is set automatically
LOG_SOLVER        | info    | CVODE maximum integration order 5
LOG_SOLVER        | info    | CVODE maximum number of nonlinear convergence failures permitted during one step 10
LOG_SOLVER        | info    | CVODE BDF stability limit detection algorithm OFF
LOG_ASSERT        | warning | [C:/Users/adrpo33/AppData/Roaming/.openmodelica/libraries/Modelica 4.0.0+maint.om/Fluid/Interfaces.mo:1027:9-1032:39:writable]
|                 | |       | The following assertion has been violated at time 0.000000
|                 | |       | ((pipe.flowModel.m_flows[1] >= 0.0 and pipe.flowModel.m_flows[1] <= 1e5)) --> "Variable violating min/max constraint: 0.0 <= pipe.flowModel.m_flows[1] <= 1e5, has value: -141.424"
info:    Result file: model_res.mat (bufferSize=10)
error:   [fmi2logger] c:/temp/model-tvf7xlol/temp/0001_PumpingSystem (logStatusError): C:/Users/adrpo33/AppData/Roaming/.openmodelica/libraries/Modelica 4.0.0+maint.om/Media/Water/IF97_Utilities.mo:1642: IF97 medium function g1: the input pressure (= 2.49291e+09 Pa) is higher than 100 MPa
```

**Comment by Primkins** [comment]
We have tested this FMU using Python-V3.x with OMSimulator. Attached the sample code to see this issue.
[FMU-OMSimulator-Issue.zip](https://github.com/user-attachments/files/17897315/FMU-OMSimulator-Issue.zip)

**Comment by lochel** [comment]
I generated a linux version of the fmu and run it with valgrind:

```
LOG_ASSERT        | warning | [/home/lochel/.openmodelica/libraries/Modelica 4.0.0+maint.om/Fluid/Interfaces.mo:1027:9-1032:39:writable]
|                 | |       | The following assertion has been violated at time 0.000000
|                 | |       | ((pipe.flowModel.m_flows[1] >= 0.0 and pipe.flowModel.m_flows[1] <= 1e5)) --> "Variable violating min/max constraint: 0.0 <= pipe.flowModel.m_flows[1] <= 1e5, has value: -141.424"
info:    Result file: model_res.mat (bufferSize=10)
error:   [fmi2logger] /home/lochel/issues/omc_13272/model-ejudl3dt/temp/0001_PumpingSystem (logStatusError): /home/lochel/.openmodelica/libraries/Modelica 4.0.0+maint.om/Media/Water/IF97_Utilities.mo:1642: IF97 medium function g1: the input pressure (= 2.49291e+09 Pa) is higher than 100 MPa
==23617== Invalid read of size 4
==23617==    at 0x4E89205: longjmp (longjmp.c:32)
==23617==    by 0xA05F035: omc_assert_fmi (in /tmp/fmi4c_OElnHD/binaries/linux64/PumpingSystem.so)
==23617==    by 0xA086360: omc_Modelica_Media_Water_IF97__Utilities_BaseIF97_Basic_g1 (in /tmp/fmi4c_OElnHD/binaries/linux64/PumpingSystem.so)
==23617==    by 0xA07F118: omc_Modelica_Media_Water_IF97__Utilities_waterBaseProp__pT (in /tmp/fmi4c_OElnHD/binaries/linux64/PumpingSystem.so)
==23617==    by 0xA024D9C: PumpingSystem_eqFunction_458 (in /tmp/fmi4c_OElnHD/binaries/linux64/PumpingSystem.so)
==23617==    by 0xA028569: PumpingSystem_functionODE (in /tmp/fmi4c_OElnHD/binaries/linux64/PumpingSystem.so)
==23617==    by 0xA074B7D: internalGetEventIndicators (in /tmp/fmi4c_OElnHD/binaries/linux64/PumpingSystem.so)
==23617==    by 0xA075B0C: fmi2DoStep (in /tmp/fmi4c_OElnHD/binaries/linux64/PumpingSystem.so)
==23617==    by 0x30EF90: fmi2_doStep (fmi4c.c:3800)
==23617==    by 0x2611A9: oms::ComponentFMUCS::stepUntil(double) (ComponentFMUCS.cpp:862)
==23617==    by 0x215126: oms::SystemWC::doStep() (SystemWC.cpp:581)
==23617==    by 0x21641A: oms::SystemWC::stepUntil(double) (SystemWC.cpp:677)
==23617==  Address 0x40 is not stack'd, malloc'd or (recently) free'd
```

**Comment by lochel** [comment]
After closely examining the coverage reports, it appears that the FMU can be simulated using FMPy, which suggests there is an issue with the fmi2 call sequence inside OMSimulator. I've opened this issue to track the progress: OpenModelica/OMSimulator#1382

Besides that, there still might be problems with fmi export.

**Comment by lochel** [comment]
@Primkins The issue should be resolved with OpenModelica/OMSimulator#1383. If the problem persists, please feel free to reopen this issue or create a new one.

Below is the valgrind summary:

```
==91317== HEAP SUMMARY:
==91317==     in use at exit: 0 bytes in 0 blocks
==91317==   total heap usage: 1,282,946 allocs, 1,282,946 frees, 89,546,044 bytes allocated
==91317==
==91317== All heap blocks were freed -- no leaks are possible
==91317==
==91317== For lists of detected and suppressed errors, rerun with: -s
==91317== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```
